### PR TITLE
refactor: refactored to make code more manageable

### PR DESCRIPTION
**Pull request by guacamole**
- Added auth/basic_auth that contains basic auth functions
- Added telemetry for future logging and metrics purposes
- Added router to contain all the routes and register route functions

Signed-off-by: guacamole <gunjanwalecha@gmail.com>

### CODE REVIEW

File: auth/signup.go [hunk 1]
```diff
@@ -21,13 +22,13 @@ type User struct {
 
 const UserNameSpace = "users"
 
-func (a *auth) ValidateUser(u User) error {
+func (u *User) Validate(store cache.Store) error {
```
Comment (jay-dee7) [inline auth/signup.go:25]: @guacamole can we call this something like `checkForDuplicatesAndValidate()`?

Comment (guacamole) [reply auth/signup.go:25]: Hey @jay-dee7 , I think ValidateUser is good as for keeping it short. we can maybe extract the checking duplicate part and merge it in a future PR? LMK

Comment (jay-dee7) [reply auth/signup.go:25]: sure that works too

**Review by jay-dee7 — APPROVED** [review]
LGTM 🥇  thanks for making it this neat. Looks much much better now. muchas gracias

**Pull request merged**